src/util/DebuggableDeferredPromise.ts

```diff
@@ -0,0 +1,20 @@
+import {DEFERRED_PROMISE_DEBUG_TIMEOUT} from '../environment.js';
+import {DeferredPromise, createDeferredPromise} from './DeferredPromise.js';
+
+/**
+ * Creates and returns a deferred promise using DEFERRED_PROMISE_DEBUG_TIMEOUT
+ * if it's specified or a normal deferred promise otherwise.
+ *
+ * @internal
+ */
+export function createDebuggableDeferredPromise<T>(
+  message: string
+): DeferredPromise<T> {
+  if (DEFERRED_PROMISE_DEBUG_TIMEOUT > 0) {
+    return createDeferredPromise({
+      message,
+      timeout: DEFERRED_PROMISE_DEBUG_TIMEOUT,
+    });
+  }
+  return createDeferredPromise();
```

Comment:
Don't we want to keep the message for default timeout?
```suggestion
  return createDeferredPromise({message});
```

Reply:
Nope, there should be no timeout and, thus, no message if it's not running the debug mode.